djvm/src/main/kotlin/net/corda/djvm/analysis/AnalysisConfiguration.kt

```diff
@@ -384,19 +398,177 @@ class AnalysisConfiguration private constructor(
                 }
             }.withBody()
              .build()
+        ).mapByClassName() + listOf(
+            object : MethodBuilder(
+                access = ACC_PRIVATE or ACC_STATIC,
+                className = sandboxed(JRELocaleProviderAdapter::class.java),
+                memberName = "isNonENLangSupported",
+                descriptor = "()Z"
+            ) {
+                override fun writeBody(emitter: EmitterModule) = with(emitter) {
+                    pushFalse()
+                    returnInteger()
+                }
+            }.withBody()
+             .build()
+        ).mapByClassName() + listOf(
+            // Create factory function to wrap java.io.InputStream.
+            object : MethodBuilder(
+                access = ACC_PUBLIC or ACC_STATIC,
+                className = sandboxed(InputStream::class.java),
+                memberName = "toDJVM",
+                descriptor = "(Ljava/io/InputStream;)Lsandbox/java/io/InputStream;"
+            ) {
+                override fun writeBody(emitter: EmitterModule) = with(emitter) {
+                    val doWrap = Label()
```

Comment:
I am missing a link here. `Currency` will probably need to load a data file using `FileInputStream`, which will not be possible inside the sandbox. So here we provide `DJVMInputStream` wrapping a generic input stream and a `toDJVM` function, but how is the `Currency` class able to invoke it?

Reply:
The `currency.data` file lives inside `deterministic-rt.jar` and is loaded using `DJVM.getSystemResourceAsStream`.